File: src/api/voice.ts
import { call } from './_core';
import type { TTSVoice, WhisperStatus } from '../types';

export const voiceApi = {
  listTTSVoices: (): Promise<TTSVoice[]> =>
    call<TTSVoice[]>('list_tts_voices'),

  synthesizeSpeech: (text: string, voiceId: string, rate: number, pitch: number): Promise<number[]> =>
    call<number[]>('synthesize_speech', { text, voiceId, rate, pitch }),

  transcribeAudio: (audioData: number[]): Promise<{ text: string; language: string }> =>
    call<{ text: string; language: string }>('transcribe_audio', { audioData }),

  getWhisperStatus: (): Promise<WhisperStatus> =>
    call<WhisperStatus>('get_whisper_status'),

  downloadWhisper: (): Promise<void> =>
    call<void>('download_whisper'),
};

File: src-tauri/src/commands/voice.rs
use crate::tts_engine::{TTSEngine, TTSRequest, TTSVoice};
use crate::whisper_engine::{TranscriptionResult, WhisperEngine, WhisperStatus};

#[tauri::command]
pub fn list_tts_voices() -> Vec<TTSVoice> {
    TTSEngine::get_voices()
}

#[tauri::command]
pub async fn synthesize_speech(text: String, voice_id: String, rate: f32, pitch: f32) -> Result<Vec<u8>, String> {
    TTSEngine::synthesize(TTSRequest { text, voice_id, rate, pitch }).await
}

#[tauri::command]
pub async fn transcribe_audio(audio_data: Vec<u8>, window: tauri::Window) -> Result<TranscriptionResult, String> {
    WhisperEngine::transcribe(audio_data, window).await
}

#[tauri::command]
pub fn get_whisper_status() -> WhisperStatus {
    WhisperEngine::get_status()
}

#[tauri::command]
pub async fn download_whisper(window: tauri::Window) -> Result<(), String> {
    WhisperEngine::download_all(window).await
}
